tests/queries/0_stateless/01881_join_on_conditions_merge.sql.j2

```diff
@@ -72,8 +72,8 @@ SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.key; -- { serverErro
 SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.key; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT, INVALID_JOIN_ON_EXPRESSION }
 SELECT * FROM t1 JOIN t2_nullable as t2 ON t2.key == t2.key2 AND (t1.id == t2.id OR isNull(t2.key2)); -- { serverError INVALID_JOIN_ON_EXPRESSION }
 SELECT * FROM t1 JOIN t2 ON t2.key == t2.key2 OR t1.id == t2.id; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 JOIN t2 ON (t2.key == t2.key2 AND (t1.key == t1.key2 AND t1.key != 'XXX' OR t1.id == t2.id)) AND t1.id == t2.id; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 JOIN t2 ON t2.key == t2.key2 AND t1.key == t1.key2 AND t1.key != 'XXX' AND t1.id == t2.id OR t2.key == t2.key2 AND t1.id == t2.id AND t1.id == t2.id; -- { serverError NOT_IMPLEMENTED }
+SELECT * FROM t1 JOIN t2 ON (t2.key == t2.key2 AND (t1.key == t1.key2 AND t1.key != 'XXX' OR t1.id == t2.id)) AND t1.id == t2.id; -- { serverError NOT_IMPLEMENTED }
```

Comment:
The following join clauses is generated:
```
Clause1:
	left_key_nodes: 
		__table1.id Int32
	right_key_nodes: 
		__table2.id Int32 
	left_condition_nodes:
		and(
			equals(
				__table1.key,
				__table1.key2
			),
			notEquals(
				__table1.key,
				'XXX'_String
			)
		) UInt8
	right_condition_nodes:
		equals(
			__table2.key,
			__table2.key2
		) UInt8,

Clause2:
	left_key_nodes:
			__table1.id Int32,
			__table1.id Int32
		right_key_nodes:
			__table2.id Int32,
			__table2.id Int32
	right_condition_nodes:
		equals(
			__table2.key,
			__table2.key2
		) UInt8, 

```

I think these are valid, but it is not implemented by `partial_merge`: `Code: 48. DB::Exception: Only `hash` join supports multiple ORs for keys in JOIN ON section. (NOT_IMPLEMENTED)`